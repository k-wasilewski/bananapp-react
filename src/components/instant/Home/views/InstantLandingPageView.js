import logo from '../../../../jpg/logo0.jpg';
import logoTitle from '../../../../jpg/logotitle.jpg';
import {Link} from 'react-router-dom';
import React, {Component} from 'react';

class InstantLandingPageView extends Component {
    render() {
        return (
            <div className='App'>
                <header className='App-header'>
                    <div className='App-break'/>
                    <img src={logo} className='App-logo' alt='logo'/>
                    <div className='App-break'/>
                    <img src={logoTitle} alt='logo-title'/>
                    {this.props.logoutMessage}
                    <button variant='outlined' onClick={this.props.doRegister}>
                        Register
                    </button>
                    {this.props.formRegister}
                    <button variant='outlined' onClick={this.props.doLogin}>
                        Log-in
                    </button>
                    {this.props.formLogin}
                    <input type='file' name='avatar' onChange={this.props.fileChangedHandler}/>
                    <button type='button' onClick={this.props.submitLoading}> Upload</button>
                    {this.props.loadingComponent}
                    {this.props.errorMsg}
                    {this.props.imagePreview}
                    <div className='App-break'/>
                    <Link to='/about'>
                        <button variant='outlined'>
                            About
                        </button>
                    </Link>
                </header>
            </div>
        );
    }
}

export default InstantLandingPageView;