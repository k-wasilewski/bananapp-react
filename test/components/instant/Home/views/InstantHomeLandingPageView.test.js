import React from 'react';
import renderer from 'react-test-renderer';
import store from "../../../../../src/redux/store";
import {BrowserRouter} from "react-router-dom";
import InstantLandingPageView from "../../../../../src/components/instant/Home/views/InstantLandingPageView";
import {Provider} from "react-redux";

describe("InstantHomeLandingPageView rendering specification", () => {
    it('InstantLandingPageView is rendered', () => {
        const component = renderer.create(
            <Provider store={store}>
                <BrowserRouter>
                    <InstantLandingPageView />
                </BrowserRouter>
            </Provider>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});