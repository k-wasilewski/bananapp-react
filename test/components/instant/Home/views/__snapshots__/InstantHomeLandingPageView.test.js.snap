// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstantHomeLandingPageView rendering specification InstantLandingPageView is rendered 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <div
      className="App-break"
    />
    <img
      alt="logo"
      className="App-logo"
      src={Object {}}
    />
    <div
      className="App-break"
    />
    <img
      alt="logo-title"
      src={Object {}}
    />
    <button
      variant="outlined"
    >
      Register
    </button>
    <button
      variant="outlined"
    >
      Log-in
    </button>
    <input
      name="avatar"
      type="file"
    />
    <button
      type="button"
    >
       Upload
    </button>
    <div
      className="App-break"
    />
    <a
      href="/about"
      onClick={[Function]}
    >
      <button
        variant="outlined"
      >
        About
      </button>
    </a>
  </header>
</div>
`;
